libxlsxwriter build failure

### Operating system

Windows x64

### Compiler

MSVC

### Steps to reproduce the behavior

```Shell
install libxslxwriter.
We use vcpkg in manifest mode with many other packages, but these build fine.
```


### Failure logs

[1/2] "C:/dev/tools/CMake/bin/cmake.exe" -E chdir "../../x64-windows-dbg" "C:/dev/tools/CMake/bin/cmake.exe" "C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/buildtrees/libxlsxwriter/src/EASE_1.1.5-df98c2d7df.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/packages/libxlsxwriter_x64-windows/debug" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DWINDOWSSTORE=OFF" "-DCMAKE_MAKE_PROGRAM=C:/dev/tools/Microsoft Visual Studio/2022/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DCMAKE_SYSTEM_VERSION=10.0.19041.0" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v143" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=/Zc:__cplusplus" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg_installed" "-DVCPKG_MANIFEST_INSTALL=OFF"
FAILED: ../../x64-windows-dbg/CMakeCache.txt 
"C:/dev/tools/CMake/bin/cmake.exe" -E chdir "../../x64-windows-dbg" "C:/dev/tools/CMake/bin/cmake.exe" "C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/buildtrees/libxlsxwriter/src/EASE_1.1.5-df98c2d7df.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/packages/libxlsxwriter_x64-windows/debug" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DWINDOWSSTORE=OFF" "-DCMAKE_MAKE_PROGRAM=C:/dev/tools/Microsoft Visual Studio/2022/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DCMAKE_SYSTEM_VERSION=10.0.19041.0" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v143" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=/Zc:__cplusplus" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg_installed" "-DVCPKG_MANIFEST_INSTALL=OFF"
-- The C compiler identification is MSVC 19.36.32532.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/dev/tools/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- The CXX compiler identification is MSVC 19.36.32532.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/dev/tools/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning (dev) at C:/dev/tools/CMake/share/cmake-3.26/Modules/FindZLIB.cmake:88 (message):
  ZLIB does not provide any COMPONENTS.  Calling

    find_package(ZLIB COMPONENTS ...)

  will always fail.
Call Stack (most recent call first):
  C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg_installed/x64-windows/share/zlib/vcpkg-cmake-wrapper.cmake:12 (_find_package)
  C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/scripts/buildsystems/vcpkg.cmake:806 (include)
  CMakeLists.txt:222 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at C:/dev/tools/CMake/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find ZLIB (missing: 1.0) (found version "1.2.12")
Call Stack (most recent call first):
  C:/dev/tools/CMake/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  C:/dev/tools/CMake/share/cmake-3.26/Modules/FindZLIB.cmake:200 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg_installed/x64-windows/share/zlib/vcpkg-cmake-wrapper.cmake:12 (_find_package)
  C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/scripts/buildsystems/vcpkg.cmake:806 (include)
  CMakeLists.txt:222 (find_package)


-- Configuring incomplete, errors occurred!
[2/2] "C:/dev/tools/CMake/bin/cmake.exe" -E chdir ".." "C:/dev/tools/CMake/bin/cmake.exe" "C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/buildtrees/libxlsxwriter/src/EASE_1.1.5-df98c2d7df.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/packages/libxlsxwriter_x64-windows" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DWINDOWSSTORE=OFF" "-DCMAKE_MAKE_PROGRAM=C:/dev/tools/Microsoft Visual Studio/2022/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DCMAKE_SYSTEM_VERSION=10.0.19041.0" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v143" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=/Zc:__cplusplus" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg_installed" "-DVCPKG_MANIFEST_INSTALL=OFF"
FAILED: ../CMakeCache.txt 
"C:/dev/tools/CMake/bin/cmake.exe" -E chdir ".." "C:/dev/tools/CMake/bin/cmake.exe" "C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/buildtrees/libxlsxwriter/src/EASE_1.1.5-df98c2d7df.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/packages/libxlsxwriter_x64-windows" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DWINDOWSSTORE=OFF" "-DCMAKE_MAKE_PROGRAM=C:/dev/tools/Microsoft Visual Studio/2022/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DCMAKE_SYSTEM_VERSION=10.0.19041.0" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v143" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=/Zc:__cplusplus" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg_installed" "-DVCPKG_MANIFEST_INSTALL=OFF"
-- The C compiler identification is MSVC 19.36.32532.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/dev/tools/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- The CXX compiler identification is MSVC 19.36.32532.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/dev/tools/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning (dev) at C:/dev/tools/CMake/share/cmake-3.26/Modules/FindZLIB.cmake:88 (message):
  ZLIB does not provide any COMPONENTS.  Calling

    find_package(ZLIB COMPONENTS ...)

  will always fail.
Call Stack (most recent call first):
  C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg_installed/x64-windows/share/zlib/vcpkg-cmake-wrapper.cmake:12 (_find_package)
  C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/scripts/buildsystems/vcpkg.cmake:806 (include)
  CMakeLists.txt:222 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at C:/dev/tools/CMake/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find ZLIB (missing: 1.0) (found version "1.2.12")
Call Stack (most recent call first):
  C:/dev/tools/CMake/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  C:/dev/tools/CMake/share/cmake-3.26/Modules/FindZLIB.cmake:200 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg_installed/x64-windows/share/zlib/vcpkg-cmake-wrapper.cmake:12 (_find_package)
  C:/dev/vision_git/builds/visum/vcpkg_build/vcpkg/scripts/buildsystems/vcpkg.cmake:806 (include)
  CMakeLists.txt:222 (find_package)


-- Configuring incomplete, errors occurred!
ninja: build stopped: subcommand failed.


### Additional context

The problem seems to be:
line 222 in CMakeLists.txt:
find_package(ZLIB REQUIRED "1.0")
since zlib does not offer a zlib.cmake, CMake uses the FindZLib module, which interprets the 1.0 as a component to search and fails. Removing the "REQUIRED 1.0", e.g., through a custom port, works.

--- a/ports/libxlsxwriter/fix-zlib-ver-required.patch
+++ b/ports/libxlsxwriter/fix-zlib-ver-required.patch
@@ -1,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 776c88a..ccc057d 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -219,7 +219,7 @@ enable_language(CXX)
+ list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+ 
+ # ZLIB
+-find_package(ZLIB REQUIRED "1.0")
++find_package(ZLIB "1.0" REQUIRED)
+ list(APPEND LXW_PRIVATE_INCLUDE_DIRS ${ZLIB_INCLUDE_DIRS})
+ message("zlib version: " ${ZLIB_VERSION})
+ 

--- a/ports/libxlsxwriter/portfile.cmake
+++ b/ports/libxlsxwriter/portfile.cmake
@@ -6,6 +6,7 @@ vcpkg_from_github(
     HEAD_REF master
     PATCHES
         dependencies.diff
+        fix-zlib-ver-required.patch
 )
 file(REMOVE_RECURSE "${SOURCE_PATH}/third_party/minizip")
 

--- a/ports/libxlsxwriter/vcpkg.json
+++ b/ports/libxlsxwriter/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "libxlsxwriter",
   "version": "1.1.5",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Libxlsxwriter is a C library that can be used to write text, numbers, formulas and hyperlinks to multiple worksheets in an Excel 2007+ XLSX file.",
   "homepage": "https://github.com/jmcnamara/libxlsxwriter",
   "license": null,

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4814,7 +4814,7 @@
     },
     "libxlsxwriter": {
       "baseline": "1.1.5",
-      "port-version": 1
+      "port-version": 2
     },
     "libxml2": {
       "baseline": "2.10.3",

--- a/versions/l-/libxlsxwriter.json
+++ b/versions/l-/libxlsxwriter.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "a877e6dd9e9652cedcb94daa467c8fd7580bf5fb",
+      "version": "1.1.5",
+      "port-version": 2
+    },
+    {
       "git-tree": "2769eadc216de088f724efd09edce1c73f9e8b4b",
       "version": "1.1.5",
       "port-version": 1

